src/App.jsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import useLocalStorage from "use-local-storage";
+
+import { Form } from './components/Form';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+
+function App() {
+  const [todos, setTodos] = useLocalStorage('todos', []);
+  const [todoTitle, setTodoTitle] = useState('');
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    if (todoTitle.trim().length > 0) {
+      const todo = {
+        title: todoTitle,
+        id: +new Date(),
+        completed: false,
+      };
+
+      setTodos([todo, ...todos]);
+    }
+
+    setTodoTitle('');
+  };
+
+  const setStatus = (todoId) => {
+    setTodos(prevTodos => prevTodos.map((todo) => {
+      if (todo.id === todoId) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+
+      return todo;
+    }));
+  };
+
+  const destroyTodo = (todoId) => {
+    setTodos(todos.filter(todo => todo.id !== todoId));
+  };
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
```

Comment:
you can use useMemo, to get todos for rendering, you don't need both at the same time(cause obviously there is only one filter applied at one time)